src/misc.cpp

```diff
@@ -33,6 +33,7 @@ std::string fileExtension(const std::string& path){
 }
 
 std::string fileName(const std::string& path){
+// TODO: if the file path is entered with slash (valid path on windows), the filename will not be returned correctly

```

Comment:
I've made a commit to address this. It also centralises the file name creation. If you don't like it we can always modify/revert it.

Reply:
Looks good to me.
The names are a bit long and depending on the display options in folders, it might be annoying to find the file you want (cavity numbers being at the end of the file name) but users will figure it out.